examples/cpp/hello_world/CLIParser.hpp

```diff
@@ -28,20 +28,26 @@ class CLIParser
 
     CLIParser() = delete;
 
+    enum entity_kind
```

Comment:
Prefer enum class over plain enums. Also, please me explicit about the type. As an enum class, name should be capital camel cased. I'd suggest adding a namespace for the example, something like `eprosima::fastdds::examples::hello_world` so `EntityKind` does not clash with any other thing.

Reply:
Not applied